alphazero/model.py

```diff
@@ -0,0 +1,101 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class ResidualBlock(nn.Module):
+    def __init__(self, in_channels, out_channels):
+        super(ResidualBlock, self).__init__()
+        self.conv = nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=1)
+        self.batch_norm = nn.BatchNorm2d(out_channels)
+        self.relu = nn.ReLU(inplace=true)
+
+        def forward(self, x):
+            """
+			Create One Residual Block where we save the input as the residual and add it after two convolutions
+			"""
+            residual = x
+
+            """
+			first convolution
+			"""
+            out = self.conv(x)
+            out = self.batch_norm(out)
+            out = self.relu(out)
+
+            """
+			second convolution, add input after batchnorm step
+			"""
+            out = self.conv(out)
+            out = self.batch_norm(out)
+            out += residual
+            out = self.relu(out)
+
+            return out
+
+
+class Net(nn.Module):
+    def __init__(
+        self,
+        block,
+        in_channels: int,
+        out_channels: int,
+        game_play: Game,
+        residual_layers: int,
+    ):
+        super(Net, self).__init__()
+
+        self.conv1 = nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=1)
+        self.bn = nn.BatchNorm2d(out_channels)
+        self.resblock = self.make_layer(block, out_channels, residual_layers)
+        self.relu = nn.ReLU(inplace=true)
+
+        self.policyconv = nn.Conv2D(
+            out_channels, out_channels=2, kernel_size=2, stride=1
+        )
+        self.policybn = nn.BatchNorm2d(2)
+        self.policyfc = nn.Linear(
+            in_channels=out_channels,
+            out_channels=game_play.width * game_play.length + 1,
```

Comment:
The property you're looking for is `height` not length. Do you add 1 here as a "passing" move? I think that would be game dependent, I think it's possible in Go but not in TTT/C4. Maybe this should be added in `Game` as a property

Reply:
Yeah the 1 is the passing value. I'll remove it for now, we can find a way to re-implement it if we have a game where passing is a thing.